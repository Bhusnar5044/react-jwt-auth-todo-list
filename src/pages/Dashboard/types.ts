export type Props = {
    // Home prop type
    click: () => void;
};
