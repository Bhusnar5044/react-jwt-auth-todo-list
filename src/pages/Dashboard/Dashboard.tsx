import { Box } from '@components/Lib';
import Todo from '@components/Todo';
import { FC, memo } from 'react';

export const Dashboard: FC = memo(() => {
    return (
        <>
            <Box display="flex" flexDirection="column" gap="2rem">
                <Todo />
            </Box>
        </>
    );
});

Dashboard.displayName = 'Dashboard';
