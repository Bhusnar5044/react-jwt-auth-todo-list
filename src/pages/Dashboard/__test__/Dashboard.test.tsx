import { render } from '@test-utils';
import { Dashboard } from '../Dashboard';

describe('Dashboard', () => {
    it('should render properly', () => {
        const { container } = render(<Dashboard />);
        expect(container).toMatchSnapshot();
    });
});
