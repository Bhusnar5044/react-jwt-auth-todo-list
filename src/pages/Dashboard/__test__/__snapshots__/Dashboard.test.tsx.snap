// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render properly 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  gap: 2rem;
}

.c8 {
  display: inline-block;
  outline: 0;
  cursor: pointer;
  text-align: center;
  border: 0;
  padding: 7px 16px;
  min-height: 36px;
  min-width: 36px;
  color: #ffffff;
  background: #008060;
  border-radius: 4px;
  font-weight: 500;
  font-size: 14px;
  box-shadow: rgba(0, 0, 0, 0.05) 0px 1px 0px 0px,rgba(0, 0, 0, 0.2) 0px -1px 0px 0px inset;
}

.c8.disabled {
  opacity: 0.7;
  pointer-events: none;
}

.c8 :hover {
  background: #006e52;
}

.c4 {
  margin: 0;
  color: inherit;
  font-size: 2.4rem;
  font-weight: 400;
  letter-spacing: -0.04rem;
  line-height: 3.2rem;
  text-align: initial;
}

.c5 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: initial;
}

.c10 {
  margin: 0;
  color: inherit;
  font-size: 1.8rem;
  font-weight: 700;
  letter-spacing: -0.02rem;
  line-height: 2.4rem;
  text-align: initial;
}

.c13 {
  display: flex;
  width: 100%;
  align-items: center;
  user-select: none;
  padding: 0.65rem 0;
  text-align: left;
}

.c13 :first-child {
  padding-top: 0;
}

.c13 :last-child {
  padding-bottom: 0;
}

.c13 .task-item-infos {
  flex: 1;
  display: flex;
  align-items: center;
  min-width: 0;
  margin-right: 1rem;
}

.c13 .task-item-checkbox {
  flex-shrink: 0;
  margin: 0;
  margin-right: 0.75rem;
  line-height: 0;
  cursor: pointer;
}

.c13 .task-item-checkbox input {
  display: none;
}

.c13 .task-item-checkbox-el {
  position: relative;
  display: inline-flex;
  width: 22px;
  height: 22px;
  border: 2px solid #ccc;
  border-radius: 50%;
  transition: 0.3s;
}

.c13 .task-item-checkbox-el:hover {
  border-color: #999;
}

.c13 .task-item-checkbox-el::before {
  content: '';
  position: absolute;
  top: 2px;
  left: 6px;
  width: 7px;
  height: 12px;
  border-style: solid;
  border-color: white;
  border-width: 0 2px 2px 0;
  transform: rotate(65deg);
  transition: 0.3s;
}

.c13 .task-item-checkbox input:checked+.task-item-checkbox-el {
  background-color: #6bffa0;
  border-color: #6bffa0;
}

.c13 .task-item-checkbox input:checked+.task-item-checkbox-el::before {
  transform: rotate(45deg);
}

.c13 .task-item-text {
  position: relative;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 700;
  transition: 0.3s ease;
}

.c13 .task-item.done .task-item-text {
  color: #ccc;
}

.c13 .task-item .task-item-text::before {
  content: '';
  position: absolute;
  top: 50%;
  left: 0;
  width: 100%;
  max-width: 0;
  height: 1px;
  background-color: #999;
  transition: 0.3s 0.1s;
}

.c13 .task-item.done .task-item-text::before {
  max-width: 100%;
}

.c13 .task-item-remove {
  flex-shrink: 0;
  line-height: 0;
  transition: 0.2s ease;
  opacity: 0;
}

.c13 .task-item:hover .task-item-remove {
  opacity: 1;
}

.c13 .task-item-remove button {
  border-radius: 50%;
  color: #ccc;
  transition: 0.2s ease;
}

.c13 .task-item-remove button:hover {
  color: tomato;
}

.c12 {
  list-style: none;
  margin: 0;
  padding: 0;
}

.c1 {
  width: 100%;
  display: flex;
  justify-content: center;
}

.c1 button {
  background: none;
  border: none;
  padding: 0;
  margin: 0;
  outline: none;
  box-shadow: none;
  cursor: pointer;
}

.c1 img,
.c1 svg {
  display: block;
  margin: 0;
  max-width: 100%;
}

.c2 {
  position: relative;
  width: 100%;
  width: 90%;
  border-radius: 0.5rem;
  background-color: #fff;
  box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
  text-align: center;
  overflow: hidden;
}

.c3 {
  padding: 2rem;
  background-color: #ccc;
  background-image: linear-gradient(135deg, #79f1a4 10%, #0e5cad 100%);
}

.c7 {
  display: block;
  padding: 1rem;
  margin: 0;
  outline: none;
  color: #495057;
  height: 2rem;
  width: 90%;
  border: none;
  font-size: 1.4rem;
  font-weight: 700;
  padding-right: 0.75rem;
}

.c9 {
  background-color: transparent;
  color: #212529;
}

.c9 :hover {
  color: #495057;
}

.c6 {
  position: relative;
  display: flex;
  width: 100%;
  align-items: center;
  text-align: left;
  border-bottom: 1px solid #e9e9e9;
  padding-left: 1.5rem;
  padding-right: 1rem;
}

.c11 {
  position: relative;
  padding: 2rem 1.5rem;
  height: 420px;
  overflow: auto;
}

<div>
  <div
    class="c0"
    display="flex"
    flexdirection="column"
    gap="2rem"
  >
    <div
      class="c1 main"
    >
      <div
        class="c2 todo"
      >
        <div
          class="c3 todo-header"
        >
          <h3
            class="c4"
            textalign="initial"
            textcolor="inherit"
            textvariant="h3"
          >
            Todo List
          </h3>
          <h5
            class="c5"
            textalign="initial"
            textcolor="inherit"
            textvariant="h5"
          >
            You have 
            <strong>
              3
            </strong>
             of 
            <strong>
              5
            </strong>
             tasks remaining
          </h5>
        </div>
        <div
          class="c6 todo-form"
        >
          <input
            class="c7"
            placeholder="Add task..."
            type="text"
          />
          <button
            class="c8 c9"
            type="button"
          >
            <h4
              class="c10"
              textalign="initial"
              textcolor="inherit"
              textvariant="h4"
            >
              +
            </h4>
          </button>
        </div>
        <div
          class="c11 todo-body"
        >
          <ul
            class="c12 todo-list"
          >
            <li
              class="c13 task-item done"
            >
              <div
                class="task-item-infos"
              >
                <label
                  class="task-item-checkbox"
                >
                  <input
                    checked=""
                    type="checkbox"
                  />
                  <div
                    class="task-item-checkbox-el"
                  />
                </label>
                <div
                  class="task-item-text"
                >
                  Task 1
                </div>
              </div>
              <div
                class="task-item-remove"
              >
                <button
                  title="Remover item"
                >
                  <svg
                    height="21"
                    viewBox="0 0 21 21"
                    width="21"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      transform="translate(5 5)"
                    >
                      <path
                        d="m10.5 10.5-10-10z"
                      />
                      <path
                        d="m10.5.5-10 10"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="c13 task-item done"
            >
              <div
                class="task-item-infos"
              >
                <label
                  class="task-item-checkbox"
                >
                  <input
                    checked=""
                    type="checkbox"
                  />
                  <div
                    class="task-item-checkbox-el"
                  />
                </label>
                <div
                  class="task-item-text"
                >
                  Task 2
                </div>
              </div>
              <div
                class="task-item-remove"
              >
                <button
                  title="Remover item"
                >
                  <svg
                    height="21"
                    viewBox="0 0 21 21"
                    width="21"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      transform="translate(5 5)"
                    >
                      <path
                        d="m10.5 10.5-10-10z"
                      />
                      <path
                        d="m10.5.5-10 10"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="c13 task-item"
            >
              <div
                class="task-item-infos"
              >
                <label
                  class="task-item-checkbox"
                >
                  <input
                    type="checkbox"
                  />
                  <div
                    class="task-item-checkbox-el"
                  />
                </label>
                <div
                  class="task-item-text"
                >
                  Task 3
                </div>
              </div>
              <div
                class="task-item-remove"
              >
                <button
                  title="Remover item"
                >
                  <svg
                    height="21"
                    viewBox="0 0 21 21"
                    width="21"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      transform="translate(5 5)"
                    >
                      <path
                        d="m10.5 10.5-10-10z"
                      />
                      <path
                        d="m10.5.5-10 10"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="c13 task-item"
            >
              <div
                class="task-item-infos"
              >
                <label
                  class="task-item-checkbox"
                >
                  <input
                    type="checkbox"
                  />
                  <div
                    class="task-item-checkbox-el"
                  />
                </label>
                <div
                  class="task-item-text"
                >
                  Task 4
                </div>
              </div>
              <div
                class="task-item-remove"
              >
                <button
                  title="Remover item"
                >
                  <svg
                    height="21"
                    viewBox="0 0 21 21"
                    width="21"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      transform="translate(5 5)"
                    >
                      <path
                        d="m10.5 10.5-10-10z"
                      />
                      <path
                        d="m10.5.5-10 10"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="c13 task-item"
            >
              <div
                class="task-item-infos"
              >
                <label
                  class="task-item-checkbox"
                >
                  <input
                    type="checkbox"
                  />
                  <div
                    class="task-item-checkbox-el"
                  />
                </label>
                <div
                  class="task-item-text"
                >
                  Task 5
                </div>
              </div>
              <div
                class="task-item-remove"
              >
                <button
                  title="Remover item"
                >
                  <svg
                    height="21"
                    viewBox="0 0 21 21"
                    width="21"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      transform="translate(5 5)"
                    >
                      <path
                        d="m10.5 10.5-10-10z"
                      />
                      <path
                        d="m10.5.5-10 10"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
